ruby/lib/clone_handler.rb

```diff
@@ -0,0 +1,45 @@
+require 'json'
+require 'net/http'
+
+class CloneHandler
+  def initialize(username)
+    @username = username
+  end
+
+  def call
+    return unless clonable?
+
+    Kernel.system(clone_command)
+  end
+
+  private
+
+  attr_reader :username
+
+  def info_uri
+    URI("https://api.github.com/users/#{username}/repos")
+  end
+
+  def response
+    @response ||= Net::HTTP.get_response(info_uri)
+  end
+
+  def response_body
+    @response_body ||= begin
+      return unless response.code == '200'
+
+      JSON.parse(response.body)
+    end
+  end
+
+  def clone_command
+    @clone_command ||= response_body&.map { |info| "git clone #{info['clone_url']}" }&.join(' & ')
+  end
```

Comment:
This is complex. (It does a lot of jobs)

To understand this I had to go work backwards from where I know the api request is in the code and move from there to the next method and then to the next function.

`info_uri -> response -> response_body -> `
This gets me to the first step in this `clone command`.

I think code is supposed to read in the other direction

aka. some pseudo code
`call` method does

`map or while loop or iterator` that does:

`get_repos all_pages` - just leave this one as a comment instead of having a function.
 1.   `get_repos_by_page_from_api`
 2.  `clone_repos_api_result`

doing this right after each iteration step will let us start cloning repos right away instead of needing to build the whole list.